### ISSUE TITLE
Queue timeout

### ISSUE THREAD

**Issue by EmptyStreets** [ostark/craft-async-queue#7]
Hi,

I have just installed the plugin. For heavy task I constantly get this error:

    The process "/RunCloud/Packages/php72rc/bin/php craft queue/exec "11530" "300" "1" "" --verbose=1 --color=" exceeded the timeout of 300 seconds.

How do I increase or disable the timeout? Thanks.

**Comment by ostark** [comment]
Hi @EmptyStreets,

I'm not fully sure, but I think the 300s timeout comes from Yii's `Queue::$ttr` property - it's not directly related to this plugin.

https://github.com/yiisoft/yii2-queue/blob/241922f57ee7fad866d9b45f46051be95d0c1aed/src/Queue.php#L74

You can try to hook into the `PushEvent` and increase the value of `$event->ttr`.

**Comment by putyourlightson** [comment]
See https://craftcms.stackexchange.com/questions/25437/queue-exec-time/25452#25452

**Comment by ostark** [comment]
@EmptyStreets if you don't "own" the queue or Job class (aka plugin or core) you can use this in your `config/app.php`

```
<?php

return [

    //..

    'bootstrap' => [function() {

        \Craft::$app->getQueue()->ttr = 3600;

    }]
];
```
@putyourlightson I added this option to your answer on stack exchange.

**Comment by swthate** [comment]
Finally! I have been fighting that same `exceeded the timeout of 300 seconds` error for two days.

```
Craft::$app->queue->ttr(x)
```
Using that before pushing the job in my module did the trick. I was pulling my hair out trying to find out why my `php.ini` settings were being ignored.

Before I installed `craft-async-queue` I was able to run the same heavy job without this error appearing. Is this something inherent to this plugin?

Thanks!!
